### PR TITLE
Unit tests

### LINKED ISSUE #8: Add unit tests for existing app files

### PR DESCRIPTION
**Pull request by j-mai**
Add unit tests for the parsing and handlers code.

Closes #8

### CODE REVIEW

File: event/parsing/codexClient.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	return eventList
+}
+
+func buildRequest(address string, auth acquire.Acquirer) (*http.Request, error) {
+	request, err := http.NewRequest(http.MethodGet, address, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	authVal, err := auth.Acquire()
```
Comment (kcajmagic) [inline event/parsing/codexClient.go:76]: We leverage `acquire.AddAuth` to add the auth 

https://github.com/xmidt-org/bascule/blob/9991ac5c1ed97661e8d613637a7517eb57303882/acquire/auth.go#L49

File: event/parsing/codexClient.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	Dest            string            `json:"dest,omitempty"`
+	TransactionUUID string            `json:"transaction_uuid,omitempty"`
+	ContentType     string            `json:"content_type,omitempty"`
+	Metadata        map[string]string `json:"metadata"`
+	Payload         string            `json:"payload,omitempty"`
+	BirthDate       int64             `json:"birth_date"`
+	PartnerIDs      []string          `json:"partner_ids,omitempty"`
+}
+
+type ResponseEvent struct {
```
Comment (kcajmagic) [inline event/parsing/codexClient.go:36]: If this struct is not being used, let's remove it.

File: event/parsing/codexClient.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		Logger:   logger,
+		Retries:  3,
+		Interval: time.Second * 30,
+
+		// Always retry on failures up to the max count.
+		ShouldRetry:       func(error) bool { return true },
+		ShouldRetryStatus: func(code int) bool { return false },
+	}
+
+	response, err := xhttp.RetryTransactor(retryOptions, http.DefaultClient.Do)(request)
```
Comment (kcajmagic) [inline event/parsing/codexClient.go:97]: An `*http.Client` should be provided incase a user wants to override the default client behavior. This can be worried about in a different PR.

**Review by kcajmagic — CHANGES_REQUESTED** [review]
Looks good, some minor changes.

File: event/parsing/codexClient.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	Data []Event
+}
+
+// query codex for events related to a device
+func (c *CodexClient) GetEvents(device string) []Event {
+	eventList := make([]Event, 0)
+
+	request, err := buildRequest(fmt.Sprintf("%s/api/v1/device/%s/events", c.Address, device), c.Auth)
+	if err != nil {
+		logging.Error(c.logger).Log(logging.ErrorKey(), err, logging.MessageKey(), "failed to build request")
```
[the remark this answers is not in the document]
Comment (j-mai) [reply event/parsing/codexClient.go:42]: Made an issue for this, will address in different PR

File: event/parsing/codexClient.go (hunk 3 above)
Comment (j-mai) [reply event/parsing/codexClient.go:90]: Made an issue for this, will address in different PR

File: event/parsing/wrpParsers_test.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+			destination: "event:reboot/mac:123/offline",
+			expectedID:  "mac:123",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.description, func(t *testing.T) {
+			deviceID, err := GetDeviceID(destinationRegex, tc.destination)
+			assert.Equal(tc.expectedID, deviceID)
+			assert.Equal(tc.expectedErr, err)
```
Comment (kcajmagic) [inline event/parsing/wrpParsers_test.go:152]: for comparing errors it is better to use `errors.Is` incase we want to wrap the errors with information. aka `fmt.Errorf`.

**Review by kcajmagic — CHANGES_REQUESTED** [review]

**Review by kcajmagic — APPROVED** [review]
